From 627fcc8be6eb7ab9f4ddf487b06efdf10cb41124 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Fri, 19 Jul 2024 07:57:32 +0300
Subject: [PATCH] [FSW-7765] fs1: arc1 iccm initialization

with this patch, added support to initialize arc1
iccm by writing 0 to entire iccm memory as part
of cold reset sequence before any read or write
operation performed on it.

Change-Id: I0c00435fa52136b13a049bd7f6fec6cc9cc4cd9b
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 7eca8d5..49d01bb 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -24,6 +24,7 @@
 #include <string.h>
 #include "dev_uart.h"
 #include "progress_bits.h"
+#include <dev_mem_copy.h>
 
 extern const int fs1_die_type;
 
@@ -158,11 +159,14 @@
 	boot_rom_global.tpm_extend_done = cold_rst_data.tpm_validation_flag;
 }
 
-/* TODO: Add functionality once mem init feature is added to memcpy block */
-int boot_rom_mem_init(__unused uint64_t addr, __unused uint32_t size)
+int boot_rom_mem_init(uint64_t addr, uint32_t size)
 {
-	/* Empty Body */
-	return 0;
+	struct mem_copy_flag flag;
+
+	flag.mem_type = MEM_COPY_MEM_TYPE_LBW;
+	flag.copy_type = MEM_COPY_TYPE_CONSTANT;
+
+	return boot_rom_mem_copy(flag, MEM_COPY_CONSTANT_CLEAR, addr, size);
 }
 
 void boot_rom_iccm_init(void)
